Первая домашка восьмого урока второго курса - готова!

lenysi4ka25 opened the pull request:
Задача № 1

lenysi4ka25 committed:
Первая домашка восьмого урока второго курса - готова!

lenysi4ka25 committed:
Первая домашка восьмого урока второго курса - готова!
Исправила!
Пропустила в задании цену и вес

lenysi4ka25 commented:
Пропустила в задании цену и вес

lenysi4ka25 committed:
Первая домашка восьмого урока второго курса - готова!
Исправила!
Пропустила в задании цену и вес

iStemmer commented on src/Main.java:
```diff
@@ -2,23 +2,34 @@
-}
\ No newline at end of file
+
+    public static void checkproductLists (boolean product){
```
на мой взгляд это немного странный метод, лучше немного скорректировать и обернуть метод add

 public static void addProduct (Set<Product> products, Product product){
              if(!products.add(product)){
                                      throw new IllegalArgumentException("Продукт уже есть!");
              }

iStemmer commented on src/Product.java:
```diff
@@ -1,11 +1,29 @@
     }
 
+    public static void checkPriceAndAmount(Product product) {
+        if (product.price == 0 && product.amount == 0) {
```
Лучше проверку эту перенести в конструктор и выкидывать ошибку. Тогда отпадет необходимость в дополнительном методе

iStemmer commented on src/Product.java:
```diff
@@ -14,7 +32,41 @@ public void setProductName(String productName) {
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(productName, price, amount);
```
Смотрим на критерий: Например, если в списке уже есть бананы, то второй раз добавить их нельзя

то есть equals & hashcode должны считаться от названия продукта (остальные поля нас не должны волновать)

lenysi4ka25 committed:
Первая домашка восьмого урока второго курса - готова!
Исправила!
Пропустила в задании цену и вес.
Домашнее задание "2", задание № 1 - выполнила)

lenysi4ka25 commented:
Домашнее задание "2", задание № 1 - выполнила)

lenysi4ka25 committed:
Первая домашка восьмого урока второго курса - готова!
Исправила!
Пропустила в задании цену и вес.
Домашнее задание "2", задание № 1 - выполнила)
Домашнее задание "2", задание № 2 - выполнила (множество целых чисел)

iStemmer commented on src/Main.java:
```diff
@@ -1,24 +1,76 @@
+                        numberIterator.remove();
+                    }
+                }
+                System.out.println(numbers);
```
зачет

iStemmer commented on src/Main.java:
```diff
@@ -1,24 +1,76 @@
+    public static void addProduct (Set<Product> products, Product product){
+        if(!products.add(product)){
+            throw new IllegalArgumentException("Продукт уже есть!");
+        }
```
пользуйся форматированием в IDEA - комбинация клавиш Ctrl+Alt+L

iStemmer commented on src/Product.java:
```diff
@@ -1,20 +1,88 @@
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(productName);
```
зачет

iStemmer commented on src/Recipe.java:
```diff
@@ -0,0 +1,56 @@
+
+
+    public final String nameRecipe;                          //название рецепта
+    public final Set <String> structure;                    //состав, множество продуктов
```
вот тут не String - а Product должен быть
то есть в составе рецепта список продуктов (как в жизни прям)

lenysi4ka25 committed:
Первая домашка восьмого урока второго курса - готова!
Исправила!
Пропустила в задании цену и вес.
Домашнее задание "2", задание № 1 - выполнила)
Домашнее задание "2", задание № 2 - выполнила (множество целых чисел)
Исправила) - в рецептах надо добавить список Продуктов (там сейчас список Строк)

iStemmer commented on src/Product.java:
```diff
@@ -1,20 +1,88 @@
 public class Product {
 
-    private String productName;
+    static ArrayList<Product> product = new ArrayList<>();
```
получается что мы в апельсине храним авоську продуктов? нелогично. 
не должен быть список продуктов в продукте

iStemmer commented on src/Recipe.java:
```diff
@@ -0,0 +1,56 @@
+
+
+    public final String nameRecipe;                          //название рецепта
+    public final Set<String> structure;                    //состав, множество продуктов
```
а вот тут должен быть 
    public final `"Set<Product>"` structure;                    //состав, множество продуктов

iStemmer commented on src/Recipe.java:
```diff
@@ -0,0 +1,56 @@
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(/*super.hashCode(), */nameRecipe/*, structure, cost*/);
```
название и список продуктов

lenysi4ka25 committed:
Первая домашка восьмого урока второго курса - готова!
Исправила!
Пропустила в задании цену и вес.
Домашнее задание "2", задание № 1 - выполнила)
Домашнее задание "2", задание № 2 - выполнила (множество целых чисел)
Исправила!

lenysi4ka25 committed:
Первая домашка восьмого урока второго курса - готова!
Исправила!
Пропустила в задании цену и вес.
Домашнее задание "2", задание № 1 - выполнила)
Домашнее задание "2", задание № 2 - выполнила (множество целых чисел)
Исправила!
Изменила на Product, а вот как в main вывести не понимаю?

lenysi4ka25 committed:
Первая домашка восьмого урока второго курса - готова!
Исправила!
Пропустила в задании цену и вес.
Домашнее задание "2", задание № 1 - выполнила)
Домашнее задание "2", задание № 2 - выполнила (множество целых чисел)
Исправила!
Изменила на Product, а вот как в main вывести не понимаю?
Добрый вечер!
Исправила.

alekseykoshelev commented on src/Recipe.java:
```diff
@@ -0,0 +1,65 @@
+public class  Recipe {
+
+
+    public String nameRecipe;                          //название рецепта
```
Тут лучше использовать модификатор `private`

lenysi4ka25 replied:
Исправила

alekseykoshelev commented on src/Product.java:
```diff
@@ -1,20 +1,75 @@
+        this.checked = false;
+    }
+
+    public Product(String productName) {
```
Странный конструктор, он ведь ничего не делает)

lenysi4ka25 replied:
Забыла удалить данный конструктор)

lenysi4ka25 committed:
Первая домашка восьмого урока второго курса - готова!
Исправила!
Пропустила в задании цену и вес.
Домашнее задание "2", задание № 1 - выполнила)
Домашнее задание "2", задание № 2 - выполнила (множество целых чисел)
Исправила!
Изменила на Product, а вот как в main вывести не понимаю?
Добрый вечер!
Исправила.
Подкорректировала) Спасибо

lenysi4ka25 committed:
Первая домашка восьмого урока второго курса - готова!
Исправила!
Пропустила в задании цену и вес.
Домашнее задание "2", задание № 1 - выполнила)
Домашнее задание "2", задание № 2 - выполнила (множество целых чисел)
Исправила!
Изменила на Product, а вот как в main вывести не понимаю?
Добрый вечер!
Исправила.
Подкорректировала) Спасибо
Не получается сделать проверку: "Количество продукта в рецепте не может быть меньше 1.". Подскажите пожалуйста что не так?
